tensorzero-internal/src/inference/providers/anthropic.rs

```diff
@@ -543,12 +548,12 @@ impl<'a> TryFrom<&'a ContentBlock> for Option<FlattenUnknown<'a, AnthropicMessag
                     },
                 },
             ))),
-            // We don't support thought blocks being passed in from a request.
-            // These are only possible to be passed in in the scenario where the
-            // output of a chat completion is used as an input to another model inference,
-            // i.e. a judge or something.
-            // We don't think the thoughts should be passed in in this case.
-            ContentBlock::Thought(_thought) => Ok(None),
+            ContentBlock::Thought(thought) => Ok(Some(FlattenUnknown::Normal(
+                AnthropicMessageContent::Thinking {
+                    thinking: &thought.text,
+                    signature: thought.signature.as_deref(),
+                },
+            ))),
```

Comment:
we should probably return None if there is no signature here, since it can't work on Anthropic.

Reply:
I think it's better to let Anthropic error here (in case they ever start supporting thought blocks without a signature)